engine/src/flutter/flow/layers/layer_state_stack.cc

```diff
@@ -448,6 +461,27 @@ class ClipRRectEntry : public LayerStateStack::StateEntry {
   FML_DISALLOW_COPY_ASSIGN_AND_MOVE(ClipRRectEntry);
 };
 
+class ClipRSuperellipseEntry : public LayerStateStack::StateEntry {
+ public:
+  ClipRSuperellipseEntry(const DlRoundSuperellipse& clip_rsuperellipse,
+                         bool is_aa)
+      : clip_rsuperellipse_(clip_rsuperellipse), is_aa_(is_aa) {}
+
+  void apply(LayerStateStack* stack) const override {
+    stack->delegate_->clipRSuperellipse(clip_rsuperellipse_,
+                                        DlCanvas::ClipOp::kIntersect, is_aa_);
+  }
+  void update_mutators(MutatorsStack* mutators_stack) const override {
+    mutators_stack->PushClipRRect(ToApproximateSkRRect(clip_rsuperellipse_));
```

Comment:
We should file a follow-on bug about supporting these in the mutators. Since this is how we communicate with embedders on "pixel ownership", it might affect pixel tug-of-wars on the edges of these things.

Reply:
Yeah I don't understand the mutators that well. Is it only used for Skia? I'm confused by the fact that it's only support Skia shape classes. Can you be more clear on what we should do? It'd be much appreciated if you can help me writing the issue you mentioned.

Reply:
No, it is used by embedders regardless of backend. The data structures were defined as Skia objects back in the day, but embedders will still be supported for a bit now and they need to know what modifications they are subject to, such as clips and such. You've just told them "we clipped with this shape, but you will clip with this other shape" leading to discrepancies...

Reply:
This won't be a trivial fix, though, so filing a bug might be best. It would only happen if someone puts an embedded app inside a container that uses an RSE for shape and clips to it - the embedded app should also be clipped to the RSE, but this will have it clipped to an approximation of the RSE even if the rest is clipped to the actual RSE on Impeller.

Reply:
It might be as simple as defining a new MutatorStack entry type that uses the DlRSE data structure (there is nothing that requires the mutators to use Sk objects, they just haven't been converted yet).

Unfortunately, all of the embedders would need to be converted to understand the new MutatorsStack entry - not that they have anything complex to do with the data, they just all have switch statements that would be missing an entry and someone needs to wander about the source base teaching them what the new stack entry type means.

Reply:
I've created an issue https://github.com/flutter/flutter/issues/163716